### ISSUE TITLE
Xbox controllers don't conect if steam is already running

### ISSUE THREAD

**Issue by deisi** [flathub/com.valvesoftware.Steam#490]
**Game information**

All. I think its a general steam issue

**Distribution name and version where applicable**

Ubuntu 19.10

**Flatpak info**

Flatpak 1.4.3
nvidia-435-21
default
host

**Problem description**

If steam is started, before the xbox controllers are connected, they will never work in big picture mode. Instead one has to turn off steam, turn on all controllers, and then turn on steam again. Hoping that no controller will ever disconnect. If any one does, one hast to repeat the process, as re-connection doesn't work. My guess is, it has something to do with the Sandboxing, but I don't know.

**Does this issue reproduce with native Steam**

Can't test as it currently doesn't run on Ubuntu 19.10, bu I think no.

**Comment by monterrr** [comment]
I've tested recently connecting one by usb cable while steam is open and it worked in game but did not test big picture..

**Comment by nanonyme** [comment]
Interesting. Might be something about the udev setup is a bit different. There's known limitations for the libudev setup inside the sandbox (eg with regards to hardware database). I don't know full implications.

**Comment by deisi** [comment]
Ah okay, I only have wireless xbox controller when I connect them I have the following in dmesg:

`[28060.190234] input: Xbox 360 Wireless Receiver as /devices/pci0000:00/0000:00:14.0/usb3/3-6/3-6:1.0/input/input39
`

**Comment by nanonyme** [comment]
Well, we do have udev. We do have hardware DB. It's a different DB than the host has. Also specifically bluetooth uses some special stuff from flatpak (there's explicit --allow=bluetooth in manifest) so it might work different than wired controllers. Sorry, can't really help much here. It might well be a more general issue.

**Comment by nanonyme** [comment]
Ah yes, there's a comment about --device=all "Which means that it cannot open joysticks and joypads that were plugged in after it was started, and needs a huge hole punched into the sandbox. So, yes, it works, for some very 1990's version of working." in that post. I guess that would explain what you're seeing, maybe?

**Comment by nanonyme** [comment]
While that is quite old at the moment, according to my understanding not much has changed since. Also looks like the proper solution doesn't work with X11 which is from Steam point of view still a big no-no.

**Comment by deisi** [comment]
But they also talk about implementing solutions from case to case. So would this mean, one would need to implement a special kind of fix just for this gamepad to work?

**Comment by nanonyme** [comment]
I think those "case to case" things are now more or less done. Pipewire is a thing, it can handle webcams. However, proper controller support is going to hard require Wayland.

**Comment by deisi** [comment]
So this will eventually be fixed on wayland?

**Comment by nanonyme** [comment]
No, because Steam and games under Steam use X11. This is unlikely to ever work well for a sandboxed Steam. inputfd and friends most likely are going to involve direct integration.

**Comment by monterrr** [comment]
I have tested using a Xbox controller via bluetooth and cable with steam already open and games play fine so this only affects big picture.

**Comment by Barraguesh** [comment]
@andreldmonteiro a little bit off topic, but how did you get your Xbox controller to connect via bluetooth?
